sku.0/sys.server/compiled/game/script/systems/buff/buff_builder_effect_map.java

```diff
@@ -0,0 +1,53 @@
+package script.systems.buff;
+
+import script.*;
+import script.library.*;
+
+import java.util.ArrayList;
+
+public class buff_builder_effect_map extends script.base_script
+{
+    public static final String DATATABLE_BUFF_BUILDER_EFFECT_MAP = "datatables/buff/buff_builder_effect_map.iff";
+
+    public buff_builder_effect_map()
+    {
+    }
+
+  
+    public static String[] getEffectList(String affectName) throws InterruptedException
+    {
+        //get the effect from the list
+
+        ArrayList<String> effectList = new ArrayList<String>();
+
+
+        int rowIndex = dataTableSearchColumnForString(affectName, "BUFF_BUILDER_AFFECTS", DATATABLE_BUFF_BUILDER_EFFECT_MAP);
+    
+        if (rowIndex >= 0)
+        {
+            int rowCount = dataTableGetNumRows(DATATABLE_BUFF_BUILDER_EFFECT_MAP);
+
+            for (int index = 0; index < rowCount; index++)
+            {
+                dictionary row = dataTableGetRow(DATATABLE_BUFF_BUILDER_EFFECT_MAP, index);

```

Comment:
new instances of objects should be defined OUTSIDE of a loop... then just reuse the object for each iteration.

Reply:
Understood. Will make an update when I can.